src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -1788,9 +1822,188 @@ class AlterFunctionStatementSegment(BaseSegment):
         Ref("FunctionParameterListGrammar"),
         OneOf(
             Sequence("RENAME", "TO", Ref("FunctionNameSegment")),
-            Sequence("SET", OneOf("SECURE", Ref("CommentEqualsClauseSegment"))),
-            Sequence("UNSET", OneOf("SECURE", "COMMENT")),
+            Sequence(
+                "SET",
+                OneOf(
+                    Ref("CommentEqualsClauseSegment"),
+                    Sequence(
+                        "API_INTEGRATION",
+                        Ref("EqualsSegment"),
+                        Ref("SingleIdentifierGrammar"),
+                    ),
+                    Sequence(
+                        "HEADERS",
+                        Ref("EqualsSegment"),
+                        Bracketed(
+                            Delimited(
+                                Sequence(
+                                    Ref("SingleQuotedIdentifierSegment"),
+                                    Ref("EqualsSegment"),
+                                    Ref("SingleQuotedIdentifierSegment"),
+                                ),
+                            ),
+                        ),
+                    ),
+                    Sequence(
+                        "CONTEXT_HEADERS",
+                        Ref("EqualsSegment"),
+                        Bracketed(
+                            Delimited(
+                                Ref("ContextHeadersGrammar"),
+                            ),
+                        ),
+                    ),
+                    Sequence(
+                        "MAX_BATCH_ROWS",
+                        Ref("EqualsSegment"),
+                        Ref("NumericLiteralSegment"),
+                    ),
+                    Sequence(
+                        "COMPRESSION",
+                        Ref("EqualsSegment"),
+                        Ref("CompressionTypeGrammar"),
+                    ),
+                    "SECURE",
+                    Sequence(
+                        OneOf("REQUEST_TRANSLATOR", "RESPONSE_TRANSLATOR"),
+                        Ref("EqualsSegment"),
+                        Ref("FunctionNameSegment"),
+                    ),
+                ),
+            ),
+            Sequence(
+                "UNSET",
+                OneOf(
+                    "COMMENT",
+                    "HEADERS",
+                    "CONTEXT_HEADERS",
+                    "MAX_BATCH_ROWS",
+                    "COMPRESSION",
+                    "SECURE",
+                    "REQUEST_TRANSLATOR",
+                    "RESPONSE_TRANSLATOR",
+                ),
```

Comment:
Can you really only unset one of these at a time?

What's the difference between the square brackets and braces in this:

<img width="590" alt="image" src="https://user-images.githubusercontent.com/10931297/162785438-297bdc13-3f2c-491e-a7d7-ae3d5f4c8aba.png">

Reply:
I tried unsetting multiple at a time and it did not work. From what I can tell, the brackets indicate keywords as optional, whereas curly braces indicate a grouping, with the pipes meaning 'or'. I think the brackets are next to `SET`, because there are multiple options for `SET` listed on different lines, but only one line listed for `UNSET`.

https://docs.snowflake.com/en/sql-reference/conventions.html#syntax-examples-and-text